google-cloud-bigquery/acceptance/bigquery/table_policy_test.rb

```diff
@@ -0,0 +1,94 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "bigquery_helper"
+
+describe Google::Cloud::Bigquery::Table, :policy, :bigquery do
+  let(:dataset_id) { "#{prefix}_dataset" }
+  let(:dataset) do
+    d = bigquery.dataset dataset_id
+    if d.nil?
+      d = bigquery.create_dataset dataset_id
+    end
+    d
+  end
+  let(:table_id) { "kittens" }
+  let(:table) do
+    t = dataset.table table_id
+    if t.nil?
+      t = dataset.create_table table_id do |schema|
+        schema.integer   "id",    description: "id description",    mode: :required
+      end
+    end
+    t
+  end
+  let(:roles) { ["bigquery.tables.delete", "bigquery.tables.get"] }
+  let(:role) { "roles/bigquery.dataOwner" }
+  let(:service_account) { bigquery.service.credentials.client.issuer }
+  let(:member) { "serviceAccount:#{service_account}" }
+
+  it "allows permissions to be tested and policy to be updated" do
+    permissions = table.test_iam_permissions roles
+    _(permissions).must_equal roles
+    _(permissions).must_be :frozen?
+
+    policy = table.policy # get
+    _(policy).must_be_kind_of Google::Cloud::Bigquery::Policy
```

Comment:
This is where we explore my poor understanding of Ruby idioms like namespacing.  Should this namespace be exposed as something like Google::Cloud::IAM::v1::Policy, or is that atypical for Ruby?  Or are we just validating the internals of the implementation?

Put another way, does namespace matter here or is resource structure all that matters?

Reply:
The google-cloud-ruby veneers strictly encapsulate the underlying "lower-level" generated client and its classes. In this case, the "lower-level" `Google::Apis::BigqueryV2::Policy` is encapsulated by the much more generally named `Google::Cloud::Bigquery::Policy`, with the possibility of completely changing out the underlying client without breaking changes, if desired (perhaps for a gRPC client... just for example).

Reply:
Sounds good.  My worry was if we end up  providing functionality that only works on objects/resources with a specific namespace.